### DISCUSSION TITLE
Configuring FastAPI Settings in Reflex

### DISCUSSION THREAD

**Discussion by jtsuru** [reflex-dev/reflex#3310] [Q&A]
I would like to disable the auto-generated documentation and openapi.json in FastAPI.
Typically, this can be achieved by creating an instance of FastAPI as follows:

```python
api = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)
```

However, within the `reflex.app.App` class, the FastAPI instance is created as shown below, making it difficult to configure FastAPI in the same manner:
```python
        # Set up the API.
        self.api = FastAPI()
```

Moreover,
```python
app = rx.App()
app.api.docs_url = None
app.api.redoc_url = None
app.api.openapi_url = None
```
does not work as expected.

While
```python
app = rx.App()
app.api = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)
app._add_cors()
app._add_default_endpoints()
```
currently seems to work as intended, it involves calling private methods.

Is there an official way to modify FastAPI settings, including the docs_url configuration?

**Comment by masenf** [comment]
@jtsuru i think it's just an oversight that we're trying to clean up and make more consistent.

A recent PR just enabled a user to pass their own `SocketIO` instance with custom parameters, we should allow passing a custom FastAPI object as well.

https://github.com/reflex-dev/reflex/blob/656914edefa7658c76e3d350abe7e9223db313bb/reflex/app.py#L206

I think instead of just overwriting the `self.api`, we could just have a check that does

```python
if self.api is None:
    self.api = FastAPI()
```

Then the caller would be able to pass the `api` kwarg to `rx.App` like

```python
app =  rx.App(api=FastAPI(docs_url=None, redoc_url=None, openapi_url=None))
```

I think this would work for your use case.

@jtsuru are you interested in submitting a PR for this?
